Sources/MapboxMaps/Foundation/Camera/CameraTransition.swift

```diff
@@ -0,0 +1,120 @@
+    /// NOTE: Incompatible with concurrent center animations
+    public var constantAnchor: CGPoint? {
+        set {
+            anchor.fromValue = newValue ?? anchor.fromValue
+            anchor.toValue = newValue
+        }
+        get {
+            return anchor.fromValue == anchor.toValue ? anchor.fromValue : nil
+        }
+    }
+
+    /// Represents a change to the bearing of the map.
+    public var bearing: Change<CLLocationDirection>
+
+    /// Ensures that bearing transitions are optimized to take the shortest path.
+    public var shouldOptimizeBearingPath: Bool = true
+
+    /// Represents a change to the pitch of the map.
+    public var pitch: Change<CGFloat>
+
+    /// Generic struct used to represent a change in a value from a starting point (i.e. `fromValue`) to an end point (i.e. `toValue`).
+    public struct Change<T> {
+        public var fromValue: T
+        public var toValue: T?
+
+        init(fromValue: T, toValue: T? = nil) {
+            self.fromValue = fromValue
+            self.toValue = toValue
+        }
+    }
+
+    internal init(with renderedCameraOptions: CameraOptions, initialAnchor: CGPoint) {
+
+        guard let renderedCenter = renderedCameraOptions.center,
+              let renderedZoom = renderedCameraOptions.zoom,
+              let renderedPadding = renderedCameraOptions.padding,
+              let renderedPitch = renderedCameraOptions.pitch,
+              let renderedBearing = renderedCameraOptions.bearing else {
+            fatalError("Values in rendered CameraOptions cannot be nil")
+        }
+
+        center  = Change(fromValue: renderedCenter)
+        zoom    = Change(fromValue: renderedZoom)
+        padding = Change(fromValue: renderedPadding)
+        pitch   = Change(fromValue: renderedPitch)
+        bearing = Change(fromValue: renderedBearing)
+        anchor  = Change(fromValue: initialAnchor)
+    }
+
+    internal var toCameraOptions: CameraOptions {
+        return CameraOptions(center: center.toValue,
+                             padding: padding.toValue,
+                             anchor: anchor.toValue,
+                             zoom: zoom.toValue,
+                             bearing: optimizedBearingToValue,
+                             pitch: pitch.toValue)
+    }
+
+    internal var fromCameraOptions: CameraOptions {
+        return CameraOptions(center: center.fromValue,
+                             padding: padding.fromValue,
+                             anchor: anchor.fromValue,
+                             zoom: zoom.fromValue,
+                             bearing: bearing.fromValue,
+                             pitch: pitch.fromValue)
+
+    }
+
+    internal var optimizedBearingToValue: CLLocationDirection? {
+
+        // If we should not optimize bearing transition, then return the original `toValue`
+        guard shouldOptimizeBearingPath else {
+            return bearing.toValue
+        }
```

Comment:
I don't think this belongs in this property because the property name implies that it always returns the "optimized" bearing. I'd rather see this property always do just that and move the decision about whether to use this property or to use the bearing.toValue directly to the call site.

Reply:
gotcha agreed.